### PR TITLE
ReCAPTCHA validation for Contact and Demo request

### PR DESCRIPTION
**Pull request by sowmiyasubramaniyan**

### CODE REVIEW

**Review by sandesh556 — DISMISSED** [review]

File: app/Http/Controllers/Front/PageController.php [hunk 1]
```diff
@@ -886,11 +885,58 @@ public function postContactUs(Request $request)
...
+
+        return false;
+    }
+
+    private function containsExcessivePunctuation($text)
+    {
+        if (preg_match('/!{5,}/', $text)) {
+            return true;
         }
+
```
Comment (RafficMohammed) [inline app/Http/Controllers/Front/PageController.php:911]: return (bool)preg_match('/!{5,}/', $text)

File: app/Http/Controllers/Front/PageController.php [hunk 2]
```diff
@@ -886,11 +885,58 @@ public function postContactUs(Request $request)
...
+            return response()->json(['message' => 'Your message was sent successfully. Thanks.'], 200);
         } catch (\Exception $ex) {
-            return redirect()->back()->with('fails', $ex->getMessage());
+            return response()->json(['error' => $ex->getMessage()], 500);
+        }
+    }
+
+    private function detectSpam($email, $message)
+    {
+        if ($this->containsExcessivePunctuation($message) || $this->containsExcessiveCaps($message)) {
```
Comment (RafficMohammed) [inline app/Http/Controllers/Front/PageController.php:896]: Call this $this->containsSpamKeywords($message) here itself

**Review by RafficMohammed — APPROVED** [review]

**Review by SarfrazS — APPROVED** [review]
ReCAPTCHA validation for Contact and Demo request issue fixed.
Screenshot 2024-02-02 152441
Screenshot 2024-02-02 153102

**Review by manikz5 — APPROVED** [review]
Security issue for recaptcha in contact-us and demo request is fixed.
After Fix: Page is getting refreshed and showing unauthorized response in console.
image

**Pull request merged**